### ISSUE TITLE
After the last update, the hot corner began to work poorly. (overview actions) Violations were observed after resuming from standby

### ISSUE THREAD

**Issue by Andybasicasd** [G-dH/custom-hot-corners-extended#47]
17 version works well. Version 15 works well. After updating to version 19, the "actions overview" after suspend "falls off". I've already reset GNOME settings, the problem persists.  (Manjaro Linux x86_64; 5.15.50-1-MANJARO; GNOME 42.2; CPU: 11th Gen Intel i3-1115G4 (4) @ 4.100GHz;). 

my observations: there are three computers at home. 2 have problems with manjaro, one with ubuntu 22.04 has no problems.

**Comment by G-dH** [comment]
I've tested the extension in Manjaro and all works without any problem.
What other extensions do you use?
Execute following command in terminal:
`journalctl /usr/bin/gnome-shell -fo cat`
then reproduce the issue and check the terminal for log. Show me the output, if any.

**Comment by Andybasicasd** [comment]
I am using extensions: AppIndicator and KStatusNotifierItem Support, Auto Move Windows, Bluetooth Quick Connect, Clipboard Indicator, Dash to Dock, Gnome 4x UI Improvements, Improved Workspace Indicator, Places Status Indicator, Screenshot Tool. 


`journalctl /usr/bin/gnome-shell -fo cat                          ✔ 
unable to update icon for Nextcloud
unable to update icon for TelegramDesktop
unable to update icon for MEGAsync
unable to update icon for Mullvad VPN1
Unhandled promise rejection. To suppress this warning, add an error handler to your promise chain with .catch() or a try-catch block around your await expression. Stack trace of the failed promise:
  _checkNeededProperties@/usr/share/gnome-shell/extensions/appindicatorsupport@rgcjonas.gmail.com/appIndicator.js:133:33
  _nameOwnerChanged@/usr/share/gnome-shell/extensions/appindicatorsupport@rgcjonas.gmail.com/appIndicator.js:154:18
  _emit@resource:///org/gnome/gjs/modules/core/_signals.js:114:47
  AppIndicatorsNameWatcher/this._watcherId<@/usr/share/gnome-shell/extensions/appindicatorsupport@rgcjonas.gmail.com/util.js:205:22

Custom Hot Corners - Extended: enabled
Mullvad VPN1, Impossible to lookup icon for 'Mullvad VPN1_4-panel' in path /tmp/.org.chromium.Chromium.eT3lkI
unable to update icon for Mullvad VPN1
Could not create transient scope for PID 0: GDBus.Error:org.freedesktop.systemd1.UnitExists: Unit app-gnome-org.gnome.Nautilus-0.scope already exists.
meta_window_set_stack_position_no_sync: assertion 'window->stack_position >= 0' failed`

**Comment by G-dH** [comment]
Nothing related. Try to disable other extensions and if it helps, find extension which is in conflict with CHC-E.

**Comment by Andybasicasd** [comment]
I found the cause on one of the computers. Dash to Dock has a "show overview on startup" setting. I turned it off. the problem is gone. (but on another computer this function is disabled, and there is a problem. I suspect that you also need to reset the gnome settings) shame on everything that's going on

**Comment by Andybasicasd** [comment]
after long work, the problem has not disappeared. I'm downgrading to version 17. no time to "play"...

**Comment by G-dH** [comment]
I'll let you know when I find the cause.

**Comment by Andybasicasd** [comment]
> I'll let you know when I find the cause.

ok. Thank You

**Comment by G-dH** [comment]
You meant Dash to Panel, not Dash to Dock, right? If so, check also if the option "Force Activities hot corner on primary monitor" is disabled. For me, Dash to Panel is terrible extension, CHC-E already includes workaround to avoid overriding hot corners by this extension.
But still, I wasn't successful in reproducing your issue with any settings. And it's interesting that the issue describes exactly the bug I had in the version 18 which I deleted from github and disabled on e.g.o.

**Comment by G-dH** [comment]
What returns this command?
`gsettings get org.gnome.desktop.interface enable-hot-corners`

**Comment by Andybasicasd** [comment]
> What returns this command? When the issue occurs. `gsettings get org.gnome.desktop.interface enable-hot-corners`

false

**Comment by Sfinx** [comment]
same here - stock ubuntu 22.04. after simple gnome-session screensaver idle blank/unblank (I do not use nor PM nor DPMS at workstation) the custom host corners stop working reliably - it works may be one time from 5-7. The bug do not related if someone using Xorg or wayland but it is related to screensaver - gnome-session embedded screensaver bugs but xscreensaver (works only on Xorg) do not. The sad thing is that there no way to restart the gnome at wayland - so re-login needed for extension to work again. but until next screensaver. can supply any additional info.

**Comment by G-dH** [comment]
Thanks for the report, I'm working on it.
@Sfinx  Even if you disable/enable the extension? Lately I'm having issues with hot corners in fullscreen mode in Ubuntu 22.04, something sometimes blocks the triggers. I added action _Reset Hot Corners_ into the _CHC-E_ section, to which you can assign a keyboard shortcut, for case you need it update without having to re-enable the extension.

**Comment by Sfinx** [comment]
> Thanks for the report, I'm working on it. @Sfinx Even if you disable/enable the extension? Lately I've been having issues with hot corners in fullscreen mode in Ubuntu 22.04, something sometimes blocks the triggers. I added action _Reset Hot Corners_ into the _CHC-E_ section, to which you can assign a keyboard shortcut, for case you need to update hot corners without having to re-enable the extension.

I do not use fullscreen at all. Re-enabling extension do not fix the bug - only re-login do. Will wait for version with Reset action, thanks !

**Comment by G-dH** [comment]
> Will wait for version with Reset action, thanks !
It's already in version 19

**Comment by G-dH** [comment]
Can you please test the latest commit?

    git clone https://github.com/G-dH/custom-hot-corners-extended.git
    cd custom-hot-corners-extended
    make install

**Comment by Sfinx** [comment]
BTW: keyboard actions work always

**Comment by G-dH** [comment]
I just can't replicate the issue on any of my systems.
One more question: You have _Use fallback hot corner triggers_ option disabled, right?

**Comment by Sfinx** [comment]
> I just can't replicate the issue on any of my systems. One more question: You have _Use fallback hot corner triggers_ option disabled, right?

Yes, disabled. I can't replicate this when using Xorg session too. Currently switched to Xorg - all is working as current git as downloaded from https://extensions.gnome.org/

**Comment by G-dH** [comment]
Now it should be OK...

**Comment by Sfinx** [comment]
> Now it should be OK...

Good news: in 943354ce4 the Reset works ! Now I can reset the extension to working state. The bad one is that now I'm seeing the bug even under X.org

**Comment by G-dH** [comment]
It's impossible! :D I was able to replicate the bug in wayland session, fixed it, tested it with the screen lock on 3 different distros with wayland, I'm using it on my own system with X11. If the reset works, then the original bug is not involved.

**Comment by Sfinx** [comment]
Can send the video ;)

**Comment by Sfinx** [comment]
BTW: No need to wait until lock - just blank. I'm doing tests with blank set to 1 minute and lock to 30 seconds after blank. And it is 100% reproducible after unblanking in 1:05 idle

**Comment by G-dH** [comment]
Super+L

**Comment by Sfinx** [comment]
Yes, but lock is only one of reasons. I've tried to investigate another ones

**Comment by G-dH** [comment]
It's quite simple, screen lock disables all extensions and unlock enables them.

**Comment by Sfinx** [comment]
https://drive.google.com/file/d/19jt5HQSAlAaTZTS3RDS8S2iVIITSw28z/view?usp=sharing

As you can see at short video:

- first ext works as moving mouse to top triggers activities overview
- the version is 20 (latest git)
- next lock/unlock
- it do not work anymore
- resetting make it back to work

This is X.org

**Comment by G-dH** [comment]
Did you check log?

**Comment by Sfinx** [comment]
What exact log ?

**Comment by G-dH** [comment]
`journalctl /usr/bin/gnome-shell -fo cat`
This command lets you watch the log in real time.

**Comment by Sfinx** [comment]
journalctl /usr/bin/gnome-shell -fo cat

```
Custom Hot Corners - Extended: enabled
Window manager warning: Overwriting existing binding of keysym 31 with keysym 31 (keycode a).
Window manager warning: Overwriting existing binding of keysym 32 with keysym 32 (keycode b).
Window manager warning: Overwriting existing binding of keysym 34 with keysym 34 (keycode d).
Window manager warning: Overwriting existing binding of keysym 33 with keysym 33 (keycode c).
Window manager warning: Overwriting existing binding of keysym 35 with keysym 35 (keycode e).
Window manager warning: Overwriting existing binding of keysym 36 with keysym 36 (keycode f).
Window manager warning: Overwriting existing binding of keysym 37 with keysym 37 (keycode 10).
Window manager warning: Overwriting existing binding of keysym 38 with keysym 38 (keycode 11).
Window manager warning: Overwriting existing binding of keysym 39 with keysym 39 (keycode 12).
```

LOCK/UNLOCK HERE


```
Timelines with detached actors are not supported. <unnamed>[<Gjs_ui_panel_AggregateMenu>:0x55c63147a620] in animation of duration 150ms but not on stage.
Custom Hot Corners - Extended: suspended
JS ERROR: Failed to initialize fprintd service: Gio.DBusError: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name net.reactivated.Fprint was not provided by any .service files
asyncCallback@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:114:23

unable to update icon for blueman
unable to update icon for indicator-solaar
unable to update icon for software-update-available
unable to update icon for livepatch
Custom Hot Corners - Extended: enabled
Window manager warning: Overwriting existing binding of keysym 32 with keysym 32 (keycode b).
Window manager warning: Overwriting existing binding of keysym 31 with keysym 31 (keycode a).
Window manager warning: Overwriting existing binding of keysym 35 with keysym 35 (keycode e).
Window manager warning: Overwriting existing binding of keysym 34 with keysym 34 (keycode d).
Window manager warning: Overwriting existing binding of keysym 36 with keysym 36 (keycode f).
Window manager warning: Overwriting existing binding of keysym 37 with keysym 37 (keycode 10).
Window manager warning: Overwriting existing binding of keysym 38 with keysym 38 (keycode 11).
Window manager warning: Overwriting existing binding of keysym 39 with keysym 39 (keycode 12).
Window manager warning: Overwriting existing binding of keysym 33 with keysym 33 (keycode c).

```

**Comment by Sfinx** [comment]
Nothing suspicious - it prints to log that "Custom Hot Corners - Extended: enabled" but it do not work (see video). Only after reset

**Comment by G-dH** [comment]
It doesn't look like stock Ubuntu. Disable all other extensions and let me know if it's really bug in CHC-E.

**Comment by G-dH** [comment]
@Andybasicasd Can you please also test the latest version from the repo?

**Comment by Andybasicasd** [comment]
> @Andybasicasd Can you please also test the latest version from the repo?

do you mean AUR?

**Comment by G-dH** [comment]
git clone https://github.com/G-dH/custom-hot-corners-extended.git
    cd custom-hot-corners-extended
    make install

**Comment by Andybasicasd** [comment]
5 times entered suspend mode... works without problems...

**Comment by G-dH** [comment]
Thanks. Then there must be another conflict in the Sfinx's system..

**Comment by G-dH** [comment]
I've made bigger code refactoring and adjusted startup timings. I hope that now it will work even for @Sfinx.

**Comment by Andybasicasd** [comment]
Thank you for the work.

**Comment by G-dH** [comment]
@Andybasicasd Thanks for the bug report and testing, I really appreciate it.

**Comment by pc00per** [comment]
Wayland is still a baby, it really needs to grow xD

**Comment by Sfinx** [comment]
> I've made a bigger code refactoring and adjusted startup timings. I hope that now it will work even for @Sfinx.

Thanks, latest git works for me - I'm still noticing a small non-working delay (~1 second) just after wake-up but it is comfortable

**Comment by G-dH** [comment]
The delay is there to avoid conflicts with other extensions. Thanks for the feedback.

**Comment by G-dH** [comment]
Version 20 is already on [extensions.gnome.org](https://extensions.gnome.org/extension/4167/custom-hot-corners-extended/), with shorter resume delay after screen lock.
Closing this issue..

**Comment by AchillesBoi** [comment]
Can we reopen this issue? I'm on Ubuntu 22.04 (GNOME 42.9) using Dash to Panel and CHC-E v28. This bug only triggers when using Dash to Panel. If I'm using Dash to Dock or nothing at all then the bug does not happen. Is Dash to Panel doing something to mess up the hot corners? If so, how can I fix it?

Conversely, if CHC-E is the culprit here what can I do to mitigate this issue?

**Comment by G-dH** [comment]
@AchillesBoi Your problem is not related to this issue. Check following settings:
Disable - `Force Activities hot corner on primary monitor` option in Dash to Panel settings
Enable - `Watch hot corner for external overrides` option in CHC-E settings

**Comment by AchillesBoi** [comment]
> @AchillesBoi Your problem is not related to this issue. Check following settings: Disable - `Force Activities hot corner on primary monitor` option in Dash to Panel settings Enable - `Watch hot corners for external overrides` option in CHC-E settings

It's 2024 and this bug still exists. Following your instructions, both options are disabled and enabled respectibly. The bug appears as soon as you log-in. Once inside, I can restart the shell with Alt+F2 then 'r' and the bug is gone, but the bug comes back if I lock the computer then re-enter.

Currently on Ubuntu 22.04.

**Comment by G-dH** [comment]
@AchillesBoi  Please open a new issue and provide all the details.
